### PR TITLE
docs: link to max spans config

### PR DESCRIPTION
**Pull request by bmorelli25**
## What does this pull request do?

This PR links the APM Overview's [dropped span](https://www.elastic.co/guide/en/apm/get-started/current/transaction-spans.html#dropped-spans) documentation to each agent's `transaction_max_spans` configuration documentation (L52-L57).

I also decided to update the rest of the content on the page while I had the file open.

## Documentation preview

http://apm-server_3708.docs-preview.app.elstc.co/diff

## Screenshot

Screen Shot 2020-04-28 at 1 48 46 PM

## Related issues

Closes https://github.com/elastic/apm/issues/126.

### CODE REVIEW

File: docs/guide/apm-data-model.asciidoc [hunk 1]
```diff
@@ -14,56 +14,57 @@ Events can contain additional <<metadata,metadata>> which further enriches your
...
+Each span contains:
 
-* A `transaction.id` attribute that refers to their parent <<transactions,transaction>>.
-* A `parent.id` attribute that refers to their parent span, or their transaction.
-* start time and duration
-* name
-* type
+* A `transaction.id` attribute that refers to its parent <<transactions,transaction>>.
+* A `parent.id` attribute that refers to its parent span or transaction.
+* Its start time and duration
```
Comment (jalvz) [inline docs/guide/apm-data-model.asciidoc:28]: nitpick: the first 2 lines end with `.`, the later 3 do not (sorry :see_no_evil:)

File: docs/guide/apm-data-model.asciidoc [hunk 2]
```diff
@@ -14,56 +14,57 @@ Events can contain additional <<metadata,metadata>> which further enriches your
...
+They measure from the start to the end of an activity,
 and they can have a parent/child relationship with other spans.
 
-Agents automatically instrument a variety of libraries to capture these spans from within your application.
-In addition, you can use the Agent API for ad hoc instrumentation of specific code paths.
+Agents automatically instrument a variety of libraries to capture these spans from within your application,
+but you can also use the Agent API for custom instrumentation of specific code paths.
 
-A span contains:
+Each span contains:
```
Comment (jalvz) [inline docs/guide/apm-data-model.asciidoc:24]: There are more fields, but I assume they are not relevant here? (`trace_id`, `subtype`, `action`, `context.db`, ``context.http`, `context.destination`...)

File: docs/guide/apm-data-model.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/guide/apm-data-model.asciidoc:28]: Love the nits

File: docs/guide/apm-data-model.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/guide/apm-data-model.asciidoc:24]: I'll add a note that this doesn't include all of the fields in a span. Initially was just trying to give users a feel for what a span could be, without including an exhaustive list.

**Pull request merged**